### ISSUE TITLE
Add dynamic mapping of planets to dimid

### ISSUE THREAD

**Issue by zmaster587** [zmaster587/AdvancedRocketry#253]
Add dynamic mapping of planets to dimid, such that planets do not have to use the same dimid upon load.  This would allow for a theorically infinte number of planets as well as destruction and creation of planets dynamically without using extra ids.  Would also eliminate conflicts in existing worlds with other dimension adding mods.

**Comment by RealGrep** [comment]
Yes please, at the moment it doesn't play nice with RFTools Dimensions at all. It completely stomped my mining dimension at Id 2. It's definitely unsafe at the moment. I set the config to start at 16000 so it would avoid other ones, now, and that should work for now. I also just moved the space station dimension ID to -3 since Actually Additions seems to be using -2.

**Comment by zmaster587** [comment]
suffocation only occurs when the atmoshpere value of a planet (stored in advRocketry/temp.dat) is less than 25, you will only suffocate in the overworld if it is set less than 25 for planet 0.  If you send me the temp.dat i can probably fix that for you or you could let AR regen itself after changing the offset in config.  

Adding mods that add dimensions to a preexisting world is not really something that dynamic dimid mapping would fix anyhow, hence the existance of configs to change the dim offset, what I suggest you do is delete the advRocketry folder, make sure the AR config's dimid offset is higher than that of the highest dimensions you currently have and let it regen the planet config.

**Comment by mrtopf** [comment]
I have the same problem and deleting temp.dat did not help.

What I did was the following:

- installed AR
- launched Minecraft which then crashed on load with a dimension id conflict of -2
- remembered a discussion on discord about it and went here. 
- changed     I:minDimension=1200,     I:spaceStationId=-12
- restarted and loaded my world
- died of succocation
- removed temp.dat
- restarted and loaded my world
- died of suffocation

Mods installed with dimensions (afaik):

- AbyssalCraft
- Cavern
- XU2
- VoidCraft (this one could actually be the -2 one but I need to verify later)
- OnFurtherTales

Here is my temp.dat: https://drive.google.com/open?id=0B4Osm0iwYkrMeXhXRVlLU0EzSTA

In a new world it worked though.

**Comment by mrtopf** [comment]
just to confirm: VoidCraft adds -2. But as I can change the station id anyway that's just fyi.

**Comment by RealGrep** [comment]
Yep, I restored a backup of my world from before I ever even added AR, and still ended up suffocating in the Overworld after restarting the server. So basically similar to you, except for restoring from backup.

**Comment by zmaster587** [comment]
Did you delete the world after closing the program (on dedicated) or closing the world (single player).  That temp.dat file has nothing in it, which indicates either something broke while trying to generate planets (in which case there should be a log) or the file was open by something else already (If you're on a Windows OS).  

I'm kinda mystified because when I test planet gen schemes I'll delete just the AR folder and it regens without a problem perhaps 100s of times in a single day.

Can you post the server logs from the same run where you delete the AR folder and you suffocate in the overworld?

**Comment by mrtopf** [comment]
Yes, I am pretty sure, I closed the world/closed minecraft. As for the temp file I maybe should have checked the size myself before uploading .. 

But I will test it again this evening and delete everything before actually launching. I will also post the log.

**Comment by mrtopf** [comment]
So here are latest.log and fml-latest.log: https://gist.github.com/mrtopf/c3d25140ba7488ee69d06a2642eb666e

I don't really see anything critical regarding AR in there though.

I deleted the folder before even starting minecraft. It got created but temp.dat was empty again.
I also suffocated.

I then tried another world in that instance which I haven't been in for a while. I also wasn't in that world before editing the configs to fix the dimension ids. I did not suffocate in it but it felt like gravity was off (if that's a thing in your mod). But temp.dat was still empty. 

Not sure if that helps. When I have some time I can try to reproduce it with a new pack and adding it before/after world gen or so. As said though, a new world seemed to work fine. But ppl will probably try to just add it to an existing world.
